.github/workflows/ports_windows.yml

```diff
@@ -17,7 +17,130 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build:
+  build-vs:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [x86, x64]
+        configuration: [Debug, Release]
+        variant: [dev, standard]
+        visualstudio: ['2017', '2019', '2022']
+        include:
+        - visualstudio: '2017'
+          runner: windows-latest
+          vs_version: '[15, 16)'
+        - visualstudio: '2019'
+          runner: windows-2019
+          vs_version: '[16, 17)'
+        - visualstudio: '2022'
+          runner: windows-2022
+          vs_version: '[17, 18)'
+        exclude:
+        - variant: standard
+          configuration: Debug
+    runs-on: ${{ matrix.runner }}
+    steps:
+    - name: Install Visual Studio 2017
+      if: matrix.visualstudio == '2017'
+      run: |
+        choco install visualstudio2017buildtools
+        choco install visualstudio2017-workload-vctools
+        choco install windows-sdk-8.1
+    - uses: microsoft/setup-msbuild@v1
+      with:
+        vs-version: ${{ matrix.vs_version }}
+    - uses: actions/setup-python@v4
+      if: matrix.runner == 'windows-2019'
+      with:
+        python-version: '3.9' 
```

Comment:
Appveyor used to have 3.8,  any reason for 3.9 ?

Reply:
It matches the version installed by default  in the `windows-2022` image.